Release/src/http/client/http_client_asio.cpp

```diff
@@ -913,6 +913,8 @@ class asio_context : public request_context, public std::enable_shared_from_this
         auto client_cast(std::static_pointer_cast<asio_client>(client));
         auto connection = client_cast->m_pool.obtain(asio_connection_pool::Type::reused_connection);
         auto connection_he = std::make_shared<asio_connection_fast_ipv4_fallback>(client, fast_ipv4_fallback_delay, connection);
+        
+        auto s = request.body().streambuf().seekpos(0, std::ios::in);
```

Comment:
This is the fix for sharing clients.

The sharing client issue was exposed only in case the server closed the connection.

Here are the steps explainig what is going on.

1. We create http_request.
2. Get connection from a cache.
3. Write request. In this case request.body().streambuf() was read and read pointer moved to the end of the buffer.
4. We get a response that connection was closed
5. Recreate context with old request (the one with empty streambuf()) and connect.
6. During sending we hit an error because content-length is diffrent to the size of the buffer.